tests/test_endpoint_correction.py

```diff
@@ -170,11 +170,21 @@ def test_endpoint_correction():
     k_translation = 200.0
 
     results = []
-    for k_rotation in [0.0, 50.0, 1000.0]:
+    for i, k_rotation in enumerate([0.0, 50.0, 1000.0]):
 
         lhs_du, rhs_du, rotations, translations = endpoint_correction.estimate_delta_us(
             k_translation, k_rotation, core_idxs, core_params, beta, lhs_xs, rhs_xs
         )
+        if i == 0:
```

Comment:
Any reason not to assert deterministic for `i > 0` too?

Reply:
Only done to avoid the test taking too long. The test takes an extra ~30 seconds if you verify all of the iterations of the loop

Reply:
Thanks for clarifying! `i == 0` as implemented looks good then